src/bigint.rs

```diff
@@ -66,6 +66,24 @@ impl Mul<Sign> for Sign {
     }
 }
 
+impl<T: PrimInt + BitAnd<Output=T>> Pow<T> for Sign {
+    type Output = Sign;
+
+    #[inline]
+    fn pow(self, other: T) -> Sign {
+        if self != Minus {
+            self
+        } else if other == T::zero() {
+            Plus
```

Comment:
The check should use `other.is_zero()`, and this case should be first, because we also need `Plus` for `NoSign.pow(0)`.